mtp_api/apps/security/tests/test_views.py

```diff
@@ -1820,3 +1826,274 @@ def test_cannot_reject_an_accepted_check(self):
+                'debit_card_sender_details': self.debit_card_sender_details.id,
+                'states': [
+                    {
+                        'reason': 'they have amazing hair',
+                        'added_by': self.added_by_user.id
+                    }
+                ]
+            },
+            format='json',
+            HTTP_AUTHORIZATION=self.auth,
+        )
+        self.assertEqual(post_response.status_code, 201)
+        auto_accept_rule = post_response.json()
+        patch_response = self.client.patch(
+            reverse(
+                'security-check-auto-accept-detail',
+                args=[auto_accept_rule['id']]
+            ),
+            data={
+                'states': [
+                    {
+                        'active': False,
+                        'reason': 'they have shaved it off',
+                        'added_by': self.added_by_user.id
+                    }
+                ]
+            },
+            format='json',
+            HTTP_AUTHORIZATION=self.auth,
+        )
+        self.assertEqual(patch_response.status_code, 200)
+        actual_response = patch_response.json()
+        self.assertIn('id', list(actual_response.keys()))
+        del actual_response['id']
+        self.assertIn('created', list(actual_response.keys()))
+        del actual_response['created']
+        self.assertIn('created', list(actual_response['states'][0].keys()))
+        del actual_response['states'][0]['created']
+        self.assertIn('created', list(actual_response['states'][1].keys()))
+        del actual_response['states'][1]['created']
+        self.assertDictEqual(
+            expected_response,
+            actual_response,
+            msg=pformat(
+                list(dictdiffer.diff(expected_response, actual_response))
+            )
+        )
+        self.assertEqual(CheckAutoAcceptRule.objects.count(), 1)
+        self.assertEqual(CheckAutoAcceptRuleState.objects.count(), 2)
+        auto_accept_rule = CheckAutoAcceptRule.objects.first()
+        self.assertEqual(auto_accept_rule.get_latest_state().reason, 'they have shaved it off')
+        self.assertEqual(auto_accept_rule.get_latest_state().added_by_id, self.added_by_user.id)
+        self.assertEqual(auto_accept_rule.is_active(), False)
+
+    def test_auto_accept_rule_reactivate(self):
+        expected_response = {
+            'prisoner_profile': self.prisoner_profile.id,
+            'debit_card_sender_details': self.debit_card_sender_details.id,
+            'states': [
+                {
+                    'active': True,
+                    'reason': 'they have amazing hair',
+                    'added_by': {
+                        'first_name': self.added_by_user.first_name,
+                        'last_name': self.added_by_user.last_name,
+                        'username': self.added_by_user.username,
+                    }
+                },
```

Comment:
I'm wondering if the "when" the state changed/was created will be needed later on.

...but this is probably a problem for another ticket ☺️

Reply:
it'll certainly be needed for the auto-accept history view. as you say that is a logical iteration for the future

Reply:
It will, the `CheckAutoAcceptRuleState.created` field is serialized as part of the `CheckAutoAcceptRuleStateSerializer` and asserted for existence in the API tests.